src/main/resources/db/changesets/20231129_snapshotrequests.yaml

```diff
@@ -0,0 +1,60 @@
+databaseChangeLog:
+  - changeSet:
+      id: snapshot_requests
+      author: rjohanek
+      changes:
+        - createTable:
+            tableName: snapshot_requests
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_dataset_snapshot_requests
+                    references: dataset(id)
+                    deleteCascade: true
+              - column:
+                  name: request_name
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_research_purpose
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: snapshot_builder_request
```

Comment:
I would use a different name here. Even though the model that this contains is currently an object of type `SnapshotBuilderRequest` I'd like to think more about what concept this column represents. Maybe `snapshot_definition` or `snapshot_specification`?